aws/aws-how-to/instances/data-science-stack-on-ec2.rst

```diff
@@ -0,0 +1,144 @@
+    sudo ubuntu-drivers install
+    sudo reboot
+
+
+After reboot, check if the drivers and CUDA have been installed properly and the GPU is detected correctly:
+
+.. code::
+
+    nvidia-smi
+
+
+The output should be similar to:
+
+.. code-block:: none
+
+        +---------------------------------------------------------------------------------------+
+        | NVIDIA-SMI 535.104.05             Driver Version: 535.104.05   CUDA Version: 12.2     |
+        |-----------------------------------------+----------------------+----------------------+
+        | GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
+        | Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
+        |                                         |                      |               MIG M. |
+        |=========================================+======================+======================|
+        |   0  Tesla T4                       On  | 00000000:00:1E.0 Off |                    0 |
+        | N/A   26C    P8               9W /  70W |      2MiB / 15360MiB |      0%      Default |
+        |                                         |                      |                  N/A |
+        +-----------------------------------------+----------------------+----------------------+
+
+        +---------------------------------------------------------------------------------------+
+        | Processes:                                                                            |
+        |  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
+        |        ID   ID                                                             Usage      |
+        |=======================================================================================|
+        |  No running processes found                                                           |
+        +---------------------------------------------------------------------------------------+
+
+Note that the GPU is properly detected and CUDA has also been installed.
+
+Install MicroK8s and DSS
+------------------------
+
+Install MicroK8s:
+
+
+.. code::
+
+    sudo snap install microk8s --channel 1.28/stable --classic
+    sudo microk8s enable storage dns rbac
+    sudo microk8s enable gpu
+
+Install DSS:
+
+.. code::
+
+    sudo snap install data-science-stack --channel latest/stable
+    dss initialize --kubeconfig="$(sudo microk8s config)"
+
+Create your first Jupyter Notebook:
+
+.. code::
+
+    dss create my-tensorflow-notebook --image=kubeflownotebookswg/jupyter-tensorflow-cuda:v1.8.0
+
+
+DSS will deploy Jupyter Notebooks with Tensorflow and with CUDA enabled. It’ll use a ``clusterIP`` from the MicroK8s cluster, which will only be accessible from inside the machine for the moment.
+
+To allow outside access, change the deployment to use a ``Nodeport`` instead of a ``clusterIP`` and reconnect using a SSH tunnel:
```

Comment:
```suggestion
To allow outside access, change the deployment to use a ``Nodeport`` instead of a ``clusterIP`` and reconnect using an SSH tunnel:
```